client/src/modules/authorization/components/SignInForm.jsx

```diff
@@ -0,0 +1,219 @@
+  isFormValid = () => {
+    const { isEmailValid, isPasswordValid, signInErrorId } = this.state;
+
+    return this.setState({
+      isFormValid: isEmailValid && isPasswordValid && !signInErrorId
+    });
+  };
+
+  handleSignIn = () => {
+    const { email, password } = this.state;
+    const { signIn } = this.props;
+
+    this.setState({ pending: true });
+    this.pendingPromise = makeAbortablePromise(signIn(email, password));
+
+    return this.pendingPromise.promise
+      .then(() => this.setState({ pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const newState = { pending: false };
+
+          if (err instanceof UnauthorizedException) {
+            newState.signInErrorId =
+              'authorization.actions.sign-in.invalid-credentials';
+          } else {
+            newState.signInErrorId = 'common.something-went-wrong';
+          }
+
+          this.setState(newState);
+        }
+      });
+  };
+
+  renderSignInError = () => {
+    const { signInErrorId } = this.state;
+    const {
+      intl: { formatMessage }
+    } = this.props;
+
+    return (
+      <p className="sign-in__error">
+        <FormattedMessage
+          id={signInErrorId}
+          values={{
+            link: (
+              <Link className="sign-in__link" to="/reset-password">
+                {formatMessage({ id: 'authorization.forgot-password' })}
+              </Link>
+            )
+          }}
+        />
+      </p>
+    );
+  };
+
+  renderForgotPassword = () => {
+    const {
+      intl: { formatMessage }
+    } = this.props;
+
+    return (
+      <p className="sign-in__forgot-password">
+        <FormattedMessage
+          id="authorization.forgot-password.question"
+          values={{
+            link: (
+              <Link className="sign-in__link" to="/reset-password">
+                {formatMessage({ id: 'authorization.forgot-password' })}
+              </Link>
+            )
+          }}
+        />
+      </p>
+    );
+  };
+
+  render() {
+    const { isFormValid, pending, signInErrorId } = this.state;
+    const { onCancel } = this.props;
+    const hasSignUpFailed = signInErrorId.length > 0;
+
+    return (
+      <div className="sign-in">
+        <h1 className="sign-in__heading">
+          <FormattedMessage id="authorization.auth-box.sign-in" />
+        </h1>
+        {signInErrorId && this.renderSignInError()}
+        <form className="sign-in__form" noValidate>
+          <AuthInput
+            disabled={pending}
+            formError={hasSignUpFailed}
+            focus
+            labelId="authorization.input.email.label"
+            name="email"
+            onChange={this.onEmailChange}
+            type="text"
+            validator={this.emailValidator}
+          />
+          <AuthInput
+            disabled={pending}
+            formError={hasSignUpFailed}
+            labelId="authorization.input.password.label"
+            name="password"
```

Comment:
are these name attributes necessary on all of these inputs? 
what are the benefits of using them?

They don't have any SEO value 
https://webmasters.stackexchange.com/questions/61609/using-the-html-input-name-attribute-and-seo
http://maileohye.com/title-name-attributes-html-anchors/

Reply:
It's a custom component, not input itself. This name is used not only for input's name attribute but mainly for label, id. Maybe it's not important for SEO (name), but it's important for accessibility, screen readers, etc. It's not a big cost to use it